src/documents/barcodes.py

```diff
@@ -83,18 +89,35 @@ def barcode_reader(image: Image) -> List[str]:
     Returns a list containing all found barcodes
     """
     barcodes = []
-    # Decode the barcode image
-    detected_barcodes = pyzbar.decode(image)
 
-    if detected_barcodes:
-        # Traverse through all the detected barcodes in image
+    if scanner_type == "PYZBAR":
+        logger.debug("Scanning for barcodes using PYZBAR")
+        # Decode the barcode image
+        detected_barcodes = pyzbar.decode(image)
+
+        if detected_barcodes:
+            # Traverse through all the detected barcodes in image
+            for barcode in detected_barcodes:
+                if barcode.data:
+                    decoded_barcode = barcode.data.decode("utf-8")
+                    barcodes.append(decoded_barcode)
+                    logger.debug(
+                        f"Barcode of type {str(barcode.type)} found: {decoded_barcode}",
+                    )
+    elif scanner_type == "ZXING":
+        logger.debug("Scanning for barcodes using ZXING")
+        detected_barcodes = zxingcpp.read_barcodes(image)
         for barcode in detected_barcodes:
-            if barcode.data:
-                decoded_barcode = barcode.data.decode("utf-8")
-                barcodes.append(decoded_barcode)
+            if barcode.text:
+                barcodes.append(barcode.text)
                 logger.debug(
-                    f"Barcode of type {str(barcode.type)} found: {decoded_barcode}",
+                    f"Barcode of type {str(barcode.format)} found: {barcode.text}",
                 )
+    else:
+        logger.warn(
+            f"Invalid Barcode Scanner {scanner_type}, not scanning for barcodes!",
```

Comment:
This is good to check, but would be better handled as a system check in `paperless/checks.py`